core/src/main/java/org/testcontainers/DockerClientFactory.java

```diff
@@ -101,6 +101,19 @@ public synchronized static DockerClientFactory instance() {
         return instance;
     }
 
+    /**
+     * Checks whether Docker is accessible and {@link #client()} is able to produce a client.
+     * @return true if Docker is available, false if not.
+     */
+    public synchronized boolean isDockerAvailable() {
+        try {
+            getOrInitializeStrategy();
```

Comment:
it is better to call `client()` here, since the Docker may be available but the checks are failing
```suggestion
            client();
```